trade-finance-manager-ui/server/controllers/deals/index.js

```diff
@@ -2,79 +2,66 @@ const api = require('../../api');
-        return apiResponse.deals.map((deal) => {
-          if (item.dealId === deal._id) {
-            // eslint-disable-next-line no-param-reassign
-            deal.tfm.stage = CONSTANTS.DEAL.DEAL_STAGE.AMENDMENT_IN_PROGRESS;
-          }
-          return deal;
-        });
-      }
-      return item;
-    });
+/**
+ * Builds a query string from query parameter values
+ * @param {string} search Value for the search query parameter
+ * @param {string} sortfield Value for the sortfield query parameter
+ * @param {string} sortorder Value for the sortorder query parameter
+ * @returns {string} Query string
+ */
+const buildQueryStringFromQueryParameterValues = (search, sortfield, sortorder) => {
+  const queryParameters = [];
+  if (search) {
+    queryParameters.push(`search=${search}`);
   }
-
-  if (apiResponse && apiResponse.deals) {
-    return res.render('deals/deals.njk', {
-      heading: 'All deals',
-      deals: apiResponse.deals,
-      activePrimaryNavigation: 'all deals',
-      activeSubNavigation: 'deal',
-      user: req.session.user,
-      sortButtonWasClicked: false,
-      activeSortByField: CONSTANTS.DEALS.TFM_SORT_BY_DEFAULT.field,
-      activeSortByOrder: CONSTANTS.DEALS.TFM_SORT_BY_DEFAULT.order,
-    });
+  if (sortfield && sortorder) {
+    queryParameters.push(`sortfield=${sortfield}&sortorder=${sortorder}`);
   }
 
-  return res.redirect('/not-found');
+  let queryString = '';
+  if (queryParameters.length > 0) {
+    queryString = `?${queryParameters.join('&')}`;
+  }
+  return queryString;
 };
 
-const queryDeals = async (req, res) => {
-  let activeSortByOrder = CONSTANTS.DEALS.TFM_SORT_BY.ASCENDING;
-  let activeSortByField = '';
-  let searchString = '';
-  const { userToken } = req.session;
-  const sortButtonWasClicked = req.body?.formId === 'deals-table-sorting';
-
-  if (req.body.search) {
-    searchString = req.body.search;
+/**
+ * Makes a request to TFM API to get the deals and renders the deals page with the deals
+ * @param {Object} req The request object
+ * @param {Object} res The response object
+ * @returns {Object} The result from rendering the page
+ */
+const getDeals = async (req, res) => {
+  const pageNumber = Number(req.params.pageNumber) || 0;
+  if (pageNumber < 0) {
+    return res.redirect('/not-found');
   }
 
-  const queryParams = { searchString };
-
-  if (req.body.descending || req.body.ascending) {
-    const order = req.body?.ascending ? 'ascending' : 'descending';
-    activeSortByField = req.body[order];
-
-    queryParams.sortBy = {
-      field: activeSortByField,
-      order,
-    };
+  const { sortfield, sortorder, search } = req.query;
+  const sortBy = {
+    field: sortfield ?? CONSTANTS.DEALS.TFM_SORT_BY_DEFAULT.field,
+    order: sortorder ?? CONSTANTS.DEALS.TFM_SORT_BY_DEFAULT.order,
+  };
+  const queryParams = {
+    sortBy,
+    pagesize: CONSTANTS.DEALS.PAGE_SIZE,
+    page: pageNumber,
+  };
+  if (search) {
+    queryParams.searchString = search;
```

Comment:
Could this maybe be moved into a helper function? e.g `generateQueryParams`, would be nicer for unit tests and simpler controller

Reply:
Done :)